src/ui.h

```diff
@@ -535,4 +535,145 @@ class pointmenu_cb : public uilist_callback
 
 void kill_advanced_inv();
 
+/**
+ * Helper for typical arrow key UI list navigation with wrap-around
+ * Add 1/-1 to val, then wrap to the range [0,size)
+ */
+template<typename V, typename S>
+inline typename std::enable_if < !std::is_enum<V>::value, V >::type increment_and_wrap( V val,
+        bool inc, S size )
+{
+    if( size == 0 ) {
+        return 0;
+    }
+    if constexpr( std::is_unsigned_v<V> ) {
+        if( !inc && val == 0 ) {
+            return size ? size - 1 : 0;
+        }
+    }
+    return modulo( inc ? ( val + 1 ) : ( val - 1 ), size );
```

Comment:
These expressions (and a few similar below) seem more readable in this way, but probably a matter of taste
```suggestion
    return modulo( val + ( inc ? +1 : -1 ), size );
```